contracts/src/challengeV2/libraries/EdgeChallengeManagerLib.sol

```diff
@@ -469,6 +469,55 @@ library EdgeChallengeManagerLib {
         return (hasRival(store, edgeId) && store.edges[edgeId].length() == 1);
     }
 
+    function updateAccuTimerCache(EdgeStore storage store, bytes32 edgeId, bytes32 claimingEdgeId, uint8 numBigStepLevel) internal returns (uint64) {
+        // calculate the time unrivaled without inheritance
+        // we don't use timeAccumulated here, since it would read the cache and double-count
+        uint64 accuTimer = timeUnrivaled(store, edgeId);
+        if (store.edges[edgeId].lowerChildId != bytes32(0)) {
+            uint64 lowerTimer = timeAccumulated(store, store.edges[edgeId].lowerChildId, numBigStepLevel);
+            if (lowerTimer == type(uint64).max) {
+                return type(uint64).max;
+            }
+            uint64 upperTimer = timeAccumulated(store, store.edges[edgeId].upperChildId, numBigStepLevel);
+            if (upperTimer == type(uint64).max) {
+                return type(uint64).max;
+            }
+            accuTimer += lowerTimer < upperTimer ? lowerTimer : upperTimer;
+        } else {
+            checkClaimIdLink(store, edgeId, claimingEdgeId, numBigStepLevel);
+            uint64 claimTimer = timeAccumulated(store, claimingEdgeId, numBigStepLevel);
+            if (claimTimer == type(uint64).max) {
+                return type(uint64).max;
+            }
+            accuTimer += claimTimer;
```

Comment:
i think we want to store max. if we do want to then we should update previous `if max` checks too
```suggestion
            if (claimTimer == type(uint64).max) {
                accuTimer = type(uint64).max;
            }
            else {
                accuTimer += claimTimer;
            }
```

Reply:
fixed, although I am thinking for simplicity we might as well just use uint256 internally and clamp when downcasting to store in the edge as uint64